File: sorting/the least logic sorting algorythms known to man/bogosort.py
"""
BOGOSORT 🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣🗣
"""
import random
 
def bogo_sort(h):
    rng = range(0, len(h)-1) # not random number generator
    while any((h[i] > h[i+1]) for i in rng): # check is list sorted
        random.shuffle(h)
 


def main():
    # randomize list
    random.seed("yeah ok bro")
    a = list(range(10))
    random.shuffle(a)

    # sort it
    random.seed(None)
    bogo_sort(a)

    # print it
    print(a)


if __name__ == "__main__":
    main()

File: sorting/the least logic sorting algorythms known to man/bogosort.cpp
#include <bits/stdc++.h>
using namespace std;

bool not_sorted(int A[], int size) {
    for (int i = 0; i < size - 1; i++) {
        if (A[i] > A[i + 1]) {
            return true;
        }
    }
    return false;
}

void shuffle(int A[], int size) {
    for (int i = 0; i < size; i++) {
        swap(A[i], A[rand() % size]);
    }
}

void bogoSort(int A[], int size) {
    while (not_sorted(A, size)) {
        shuffle(A, size);
    }
}

int main() {
    // randomize array
    unsigned int size;
    int A[10];
    for (int i = 0; i < 10; i++)
        A[i] = i;
    shuffle(A, 10);

    // sort it
    bogoSort(A, 10);

    // print it
    for (auto i: A) {
        cout << i << ' ';
    }
    return 0;
}
